src/descriptor/mod.rs

```diff
@@ -66,6 +66,170 @@ pub use self::key::{
 /// public key from the descriptor.
 pub type KeyMap = HashMap<DescriptorPublicKey, DescriptorSecretKey>;
 
+/// A general trait for Bitcoin descriptor.
+/// It should also support FromStr, fmt::Display and should be liftable
+/// to bitcoin Semantic Policy.
+/// Offers function for witness cost estimation, script pubkey creation
+/// satisfaction using the [Satisfier] trait.
+// Unfortunately, the translation function cannot be added to trait
+// because of traits cannot know underlying generic of Self.
+// Thus, we must implement additional trait for translate function
+pub trait DescriptorTrait<Pk: MiniscriptKey>:
+    FromStr + Display + Debug + Clone + Eq + PartialEq + PartialOrd + Ord + Liftable<Pk>
```

Comment:
We can probably drop most (all?) of these requirements.

Reply:
I removed the bounds. But unfortunately, removing them did not help that much with code ergonomics.